gum/backend-darwin/gumprocess-darwin.c

```diff
@@ -527,6 +541,92 @@ gum_store_address_if_export_name_matches (const GumExportDetails * details,
   return TRUE;
 }
 
+void
+gum_process_enumerate_malloc_ranges (GumFoundMallocRangeFunc func,
+                                     gpointer user_data)
+{
+  task_t task;
+  kern_return_t ret;
+  unsigned i;
+  vm_address_t * malloc_zone_addresses;
+  unsigned malloc_zone_count;
+
+  task = mach_task_self ();
+
+  ret = malloc_get_all_zones (task,
+      gum_malloc_memory_reader, &malloc_zone_addresses,
+      &malloc_zone_count);
+  if (ret != KERN_SUCCESS)
+    return;
+
+  for (i = 0; i != malloc_zone_count; i++) {
+    vm_address_t zone_address;
+    malloc_zone_t * zone;
+
+    zone_address = malloc_zone_addresses[i];
+    zone = (malloc_zone_t *) zone_address;
+
+    if (zone && zone->introspect && zone->introspect->enumerator) {
+      struct _gum_malloc_helper_info info = {func, user_data, FALSE};
+
+      zone->introspect->enumerator (task, (void *) &info,
+          MALLOC_PTR_IN_USE_RANGE_TYPE, zone_address,
+          gum_malloc_memory_reader,
+          gum_process_enumerate_malloc_ranges_helper);
+
+      if (info.should_abort)
+        return;
+    }
+  }
+}
+
+static void
+gum_process_enumerate_malloc_ranges_helper (task_t task,
+                                            void * user_data,
+                                            unsigned type, vm_range_t * ranges,
+                                            unsigned count)
+{
+  struct _gum_malloc_helper_info * info;
+  GumMemoryRange gum_range;
+  GumMallocRangeDetails details;
+  unsigned i;
+
+  info = (struct _gum_malloc_helper_info *) user_data;
```

Comment:
For simple assignment from arguments we tend to put them inline with the declaration. (Sorry, I just thought of this.)